src/zenml/integrations/deepchecks/data_validators/deepchecks_data_validator.py

```diff
@@ -0,0 +1,450 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Implementation of the Deepchecks data validator."""
+
+from typing import (
+    Any,
+    ClassVar,
+    Dict,
+    Optional,
+    Sequence,
+    Tuple,
+    Type,
+    Union,
+    cast,
+)
+
+import pandas as pd
+from deepchecks.core.checks import BaseCheck
+from deepchecks.core.suite import SuiteResult
+from deepchecks.tabular import Dataset as TabularData
+from deepchecks.tabular import Suite as TabularSuite
+
+# not part of deepchecks.tabular.checks
+from deepchecks.tabular.suites import full_suite as full_tabular_suite
+from deepchecks.vision import Suite as VisionSuite
+from deepchecks.vision import VisionData
+from deepchecks.vision.suites import full_suite as full_vision_suite
+from sklearn.base import ClassifierMixin
+from torch.nn import Module  # type: ignore[attr-defined]
+from torch.utils.data.dataloader import DataLoader
+
+from zenml.data_validators import BaseDataValidator
+from zenml.environment import Environment
+from zenml.integrations.deepchecks import DEEPCHECKS_DATA_VALIDATOR_FLAVOR
+from zenml.integrations.deepchecks.validation_checks import (
+    DeepchecksDataDriftCheck,
+    DeepchecksDataIntegrityCheck,
+    DeepchecksModelDriftCheck,
+    DeepchecksModelValidationCheck,
+    DeepchecksValidationCheck,
+)
+from zenml.logger import get_logger
+from zenml.steps import STEP_ENVIRONMENT_NAME, StepEnvironment
+from zenml.utils.string_utils import random_str
+
+logger = get_logger(__name__)
+
+
+class DeepchecksDataValidator(BaseDataValidator):
+    """Deepchecks data validator stack component."""
+
+    # Class Configuration
+    FLAVOR: ClassVar[str] = DEEPCHECKS_DATA_VALIDATOR_FLAVOR
+    NAME: ClassVar[str] = "Deepchecks"
+
+    @staticmethod
+    def _split_checks(
+        check_list: Sequence[str],
+    ) -> Tuple[Sequence[str], Sequence[str]]:
+        """Split a list of check identifiers in two lists, one for tabular and one for computer vision checks.
```

Comment:
> 80 chars..

Reply:
the docstings linter prevents me from splitting this into multiple lines. with all these rules enforced, something's gotta give